### PR TITLE
Combine transaction

### PR DESCRIPTION
**Pull request by icerove**
fix issue #132 
comment: it is a little different from the designs, since every action has been written into a component that shows the whole information, if changed to what design shows, there will be many changes and create several new component to reuse and inspect for different situation. Have a look and give feedback about this

### CODE REVIEW

**Comment by kcole16** [comment]
> I like the designed version better as it does not have duplicated information which may confuse people.

@icerove I agree with @frol here. We want to communicate to the user that each action was one transaction,  by one user,  confirmed with one timestamp.

**Comment by icerove** [comment]
okay, I will do that.

**Comment by kcole16** [comment]
@icerove Just noticed some missing formatting on the [TX detail](https://near-explorer-frontend-pr-147.onrender.com/transactions/4CLfJE6PV1kuRpDPDgmk98Up5bVuuWs1isVx2vxskxWZ) and [block detail](https://near-explorer-frontend-pr-147.onrender.com/blocks/AJtxLT6cRA5WUnys78qDGKsM7PHUDEUpA8h1g634RQ3S) pages.



On these pages, we can remove the "Batch Transaction" text, timestamp, and sender (since it's listed above).



Forgot to spec this before!

**Comment by icerove** [comment]
> @icerove Just noticed some missing formatting on the [TX detail](https://near-explorer-frontend-pr-147.onrender.com/transactions/4CLfJE6PV1kuRpDPDgmk98Up5bVuuWs1isVx2vxskxWZ) and [block detail](https://near-explorer-frontend-pr-147.onrender.com/blocks/AJtxLT6cRA5WUnys78qDGKsM7PHUDEUpA8h1g634RQ3S) pages.
Make sence to TX detail page but block detail page still need batch I think

**Comment by kcole16** [comment]
> Make sence to TX detail page but block detail page still need batch I think

Good point!

**Comment by icerove** [comment]
single transaction is not correct on detail page

File: frontend/src/components/transactions/ActionsList.tsx [hunk 1]
```diff
@@ -1,31 +1,220 @@
...
+                </Col>
+                <Col md="4" xs="5" className="ml-auto text-right">
+                  <Row>
+                    <Col className="action-row-txid">
+                      <TransactionLink transactionHash={transaction.hash} />
+                    </Col>
+                  </Row>
+                  <Row>
+                    <Col className="action-row-timer">
+                      <span className="action-row-timer-status">{`Completed`}</span>{" "}
```
Comment (frol) [inline frontend/src/components/transactions/ActionsList.tsx]: Since #143 there is a proper `transaction.status` value, which can be formatted as `<ExecutionStatus status={transaction.status} />` (`import ExecutionStatus from "../utils/ExecutionStatus";`)

File: frontend/src/components/transactions/ActionsList.tsx [hunk 2]
```diff
@@ -1,31 +1,220 @@
...
+import AccountLink from "../utils/AccountLink";
+import TransactionLink from "../utils/TransactionLink";
+import Timer from "../utils/Timer";
 
 export interface Props {
   actions: (T.Action | keyof T.Action)[];
   transaction: T.Transaction;
   viewMode?: ViewMode;
   reversed?: boolean;
+  TD?: boolean;
```
Comment (frol) [inline frontend/src/components/transactions/ActionsList.tsx]: What is this?

File: frontend/src/components/transactions/ActionsList.tsx (hunk 1 above)
Comment (icerove) [reply frontend/src/components/transactions/ActionsList.tsx]: got it and change

File: frontend/src/components/transactions/ActionsList.tsx (hunk 2 above)
Comment (icerove) [reply frontend/src/components/transactions/ActionsList.tsx]: on TransactionDetail page, kendall think no more need to show detail on each transaction row, because it is already show on the title, so I add this to check whether on the detail page to just show transaction information

**Comment by kcole16** [comment]
@icerove Let's center the action descriptions here:

**Comment by icerove** [comment]
@kcole16 You mean in detail page, every action is in the center? I am not very sure what do you want?

**Comment by kcole16** [comment]
> @kcole16 You mean in detail page, every action is in the center? I am not very sure what do you want?

Ah right, I meant align the action description (e.g. "New key added") with the icon on the left, like this:

File: frontend/src/components/transactions/ActionsList.tsx [hunk 3]
```diff
@@ -1,31 +1,222 @@
...
+                      </span>{" "}
+                      <Timer time={transaction.blockTimestamp} />
+                    </Col>
+                  </Row>
+                </Col>
+              </Row>
+            </Col>
+          </Row>
+        ) : null}
+        {actionRows}
```
Comment (frol) [inline frontend/src/components/transactions/ActionsList.tsx]: I suggest nesting the batched rows in terms of HTML structure. This way you will have better control over indentation (you just set margin for the block and no need to apply `margin-left` to individual lines) and other styling.

File: frontend/src/components/transactions/ActionsList.tsx (hunk 2 above)
Comment (frol) [reply frontend/src/components/transactions/ActionsList.tsx]: 1. TD is completely broken naming. Please, never use such names. I believe, we can completely avoid this special case here.
2. Merge `ActionRowBatch` into `ActionRow` (extend `viewMode` if necessary). Please, avoid code duplication.

File: frontend/src/components/transactions/ActionsList.tsx [hunk 4]
```diff
@@ -164,9 +164,8 @@ export default class extends React.Component<Props> {
 
           .action-row-title {
             font-family: BentonSans;
-            font-size: 14px;
+            font-size: 1rem;
```
Comment (kcole16) [inline frontend/src/components/transactions/ActionsList.tsx]: Why this change? Makes the font quite a bit larger

Comment (icerove) [reply frontend/src/components/transactions/ActionsList.tsx]: because this makes the logo and text not in the same baseline and could not center

File: frontend/src/components/transactions/ActionRow.tsx [hunk 5]
```diff
@@ -40,152 +51,75 @@ export default class extends React.Component<Props, State> {
...
+                      transaction={transaction}
+                      actionKind={actionKind}
+                      actionArgs={actionArgs}
+                    />
+                  </Col>
+                </Row>
+              </Col>
+            </Row>
+          </Col>
+        </Row>
```
Comment (frol) [inline frontend/src/components/transactions/ActionRow.tsx]: There is quite a duplication with the `else` branch. Let's merge it.

File: frontend/src/components/transactions/ActionRow.tsx [hunk 6]
```diff
@@ -40,152 +51,75 @@ export default class extends React.Component<Props, State> {
...
-          }
-
-          .action-row-timer {
-            font-family: BentonSans;
-            font-size: 12px;
-            color: #999999;
-            font-weight: 100;
-          }
+    let actionRow;
+    if (detail || batch) {
```
[the remark this answers is not in the document]
Comment (icerove) [reply frontend/src/components/transactions/ActionRow.tsx]: we need two flags, batch is for batch the whole transactions and show batch line like this: 
image

but for detail page, we will only display the basic line just no details for every actions on the transaction detail page, like this: 
image

File: frontend/src/components/transactions/ActionRow.tsx (hunk 5 above)
Comment (icerove) [reply frontend/src/components/transactions/ActionRow.tsx]: I tried, but I found it is not worth to merge, because it is not actually a couple duplication, so many different place to split if we really merge them into one. See the <Col> you wrote for the format of the cell. if we just show message and no other message, what I delete for simple actionRow is the simplest.

**Comment by kcole16** [comment]
@frol Think this is ready to merge?

**Comment by frol** [comment]
I think it is ready to merge.

@icerove Please, review my changes.

**Comment by frol** [comment]
I have been considering better naming for `ActionRowBlock` and realized that we use a similar block layout for AccountRow and BlockRow, so we should consider moving `ActionRowBlock` to utils, rename it to something more generic, and reuse it everywhere.

**Comment by icerove** [comment]
I will merge it and open a new PR to make changes for `ActionRowBlock`
